tests/periph_i2c/main.c

```diff
@@ -20,270 +20,547 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <errno.h>
 
 #include "periph_conf.h"
 #include "periph/i2c.h"
 #include "shell.h"
 
+#ifndef GET_I2C_CLK_SPD
+#define GET_I2C_CLK_SPD(x)  (-1)
+#endif
+
+#ifndef I2C_ACK
+#define I2C_ACK  (0)
+#endif
+
+#define INVALID_ARGS    puts("Error: Invalid number of arguments");
+
 #define BUFSIZE        (128U)
 
 static int i2c_dev = -1;
 
-static i2c_speed_t map_speed[] = {
-    I2C_SPEED_LOW,
-    I2C_SPEED_NORMAL,
-    I2C_SPEED_FAST,
-    I2C_SPEED_FAST_PLUS,
-    I2C_SPEED_HIGH
-};
 
-int cmd_init_master(int argc, char **argv)
-{
-    int dev, speed, res;
+int check_dev(void) {
+    if (i2c_dev < 0) {
+        puts("Error: no I2C device was initialized");
```

Comment:
there is `ENODEV` which you could use here and print similarly to the other errors? See `print_i2c_error`, too

Reply:
Yup, nice catch, will do! [TODO]